spyder_kernels/customize/spydercustomize.py

```diff
@@ -393,6 +396,21 @@ def exec_code(code, filename, ns_globals, ns_locals=None, post_mortem=False,
         else:
             ast_code = ast.parse(transform_cell(code))
 
+        # Print warning for global
+        if global_warning and SHOW_GLOBAL_MSG:
+            has_global = any(
+                isinstance(node, ast.Global) for node in ast.walk(ast_code))
```

Comment:
Could this make running large files (i.e. with more than 1000 lines of code) slow?

Reply:
Slightly slower, but this shouldn't take too long. See for example:
```python
In [1]: import ast

In [2]: code = "a = 123\n" * 10000

In [3]: ast_code = ast.parse(code)

In [4]: %timeit any(isinstance(node, ast.Global) for node in ast.walk(ast_code))
47.9 ms ± 821 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
The complexity here is linear, so I am not too worried.

Reply:
Ok, that's a really small delay. Thanks for checking it.